monkestation/code/modules/modular_computers/file_system/programs/lifeline.dm

```diff
@@ -0,0 +1,193 @@
+	var/last_icon_state = ""
+
+
+/datum/computer_file/program/lifeline/on_start(mob/living/user)
+	. = ..()
+	if(.)
+		blueshield = istype(computer, /obj/item/modular_computer/pda/blueshield)
+		START_PROCESSING(SSfastprocess, src)
+		return
+	return FALSE
+
+/datum/computer_file/program/lifeline/kill_program(mob/user)
+	sensors = list()
+	selected = null
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/datum/computer_file/program/lifeline/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+/datum/computer_file/program/lifeline/ui_data(mob/user)
+	var/list/data = list()
+	data["selected"] = selected
+	data["sensors"] = update_sensors()
+	data["settings"] = list(
+		blueshield = blueshield,
+		sortAsc = sort_asc,
+		sortBy = sort_by
+	)
+	return data
+
+/datum/computer_file/program/lifeline/ui_act(action, params, datum/tgui/ui, datum/ui_state/state)
+	switch(action)
+		if("select")
+			selected = params["ref"]
+		if("sortAsc")
+			sort_asc = params["val"]
+		if("sortBy")
+			sort_by = params["val"]
+		if("blueshield")
+			blueshield = params["val"]
+	return TRUE
+
+// Copied and modified from "code\game\machinery\computer\crew.dm"
+/datum/computer_file/program/lifeline/proc/tracking_level(mob/living/tracked_living_mob, z)
+	// Check if z-level is correct
+	var/turf/pos = get_turf(tracked_living_mob)
+
+	// Is our target in nullspace for some reason?
+	if(!pos)
+		stack_trace("Tracked mob has no loc and is likely in nullspace: [tracked_living_mob] ([tracked_living_mob.type])")
+		return SENSOR_OFF
+
+	// Machinery and the target should be on the same level or different levels of the same station
+	if(pos.z != z && (!is_station_level(pos.z) || !is_station_level(z)) && !HAS_TRAIT(tracked_living_mob, TRAIT_MULTIZ_SUIT_SENSORS))
+		return SENSOR_OFF
+
+	// Set sensor level based on whether we're in the nanites list or the suit sensor list.
+	if(tracked_living_mob in GLOB.nanite_sensors_list)
+		return SENSOR_COORDS
+
+	var/mob/living/carbon/human/tracked_human = tracked_living_mob
+
+	// Check their humanity.
+	if(!ishuman(tracked_human))
+		stack_trace("Non-human mob is in suit_sensors_list: [tracked_living_mob] ([tracked_living_mob.type])")
+		return SENSOR_OFF
+
+	// Check they have a uniform
+	var/obj/item/clothing/under/uniform = tracked_human.w_uniform
+	if (!istype(uniform))
+		stack_trace("Human without a suit sensors compatible uniform is in suit_sensors_list: [tracked_human] ([tracked_human.type]) ([uniform?.type])")
+		return SENSOR_OFF
+
+	// Check if their uniform is in a compatible mode.
+	if (uniform.has_sensor >= HAS_SENSORS)
+		return uniform.sensor_mode
+	return SENSOR_OFF
+
+/datum/computer_file/program/lifeline/proc/update_sensors()
+	var/turf/pos = get_turf(computer)
+	if (world.time <= last_update_time + 3 SECONDS && sensors)
+		return sensors
+
+	sensors = list()
+	for(var/tracked_mob in GLOB.suit_sensors_list | GLOB.nanite_sensors_list)
+		var/mob/living/tracked_living_mob = tracked_mob
```

Comment:
```suggestion
	for(var/mob/living/tracked_living_mob as anything in GLOB.suit_sensors_list | GLOB.nanite_sensors_list)
```

remove the `as anything` if you want to pre-emptively exclude any null or non-living mobs

Reply:
i'll leave this as is to maintain consistency with https://github.com/Monkestation/Monkestation2.0/blob/594806ce3c97f118c1e8f535d4440bb4732821e6/code/game/machinery/computer/crew.dm#L198